lib/config/index.js

```diff
@@ -63,75 +58,75 @@ if (config.ldap.tlsca) {
   let tlsOptions = {
     ca: caContent
   }
-  config.ldap.tlsOptions = config.ldap.tlsOptions ? Object.assign(config.ldap.tlsOptions, tlsOptions) : tlsOptions
+  config.load({
+    ldap: {
+      tlsOptions: config.get(ldap).tlsOptions ? Object.assign(config.get('ldap').tlsOptions, tlsOptions) : tlsOptions
+    }
+  })
 }
 
 // Permission
-config.permission = Permission
+let permissions = Permission
 if (!config.allowAnonymous && !config.allowAnonymousEdits) {
-  delete config.permission.freely
+  delete permissions.freely
 }
-if (!(config.defaultPermission in config.permission)) {
-  config.defaultPermission = config.permission.editable
+if (!(config.defaultPermission in permissions)) {
+  config.load({defaultPermission: permissions.editable})
 }
 
-// cache result, cannot change config in runtime!!!
-config.isStandardHTTPsPort = (function isStandardHTTPsPort () {
-  return config.useSSL && config.port === 443
-})()
-config.isStandardHTTPPort = (function isStandardHTTPPort () {
-  return !config.useSSL && config.port === 80
-})()
+config.load({permission: permissions})
 
 // cache serverURL
-config.serverURL = (function getserverurl () {
-  var url = ''
-  if (config.domain) {
-    var protocol = config.protocolUseSSL ? 'https://' : 'http://'
-    url = protocol + config.domain
-    if (config.urlAddPort) {
-      if (!config.isStandardHTTPPort || !config.isStandardHTTPsPort) {
-        url += ':' + config.port
+config.load({
+  serverURL: (function getserverurl () {
```

Comment:
why is this a named function (`getserverurl`)?

Reply:
Because.

It was named like that before I took care of it, therefore I didn't change it. We could do this now, but it's also not really a problem. I actually consider to make serverURL an public option that can be used in stead of the `domain`, `protocolUseSSL` and `protocolAddPort` construct which is unneeded complicated.